src/main/java/bio/terra/service/snapshotbuilder/SnapshotRequestDao.java

```diff
@@ -134,6 +154,44 @@ public SnapshotAccessRequestResponse create(
     UUID id = keyHolder.getId();
     return getById(id);
   }
+  /**
+   * Create a new Snapshot Access Request for the given snapshot id.
+   * @param snapshotId associated with the snapshot request.
+   * @param request the snapshot access request.
+   * @param email the email of the user creating the request.
+   * @return the created snapshot access request response.
+   */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public SnapshotAccessRequestResponse createV2(
```

Comment:
I called this createV2, I figure the name doesn't matter too much since we will be removing the other create soon and then can rename this create. But, we could also call this something more specific like createFromSnapshot if we think that is better.

Reply:
If we remove the API connection for the dataset version of the endpoint, you could migrate the DAO code to use snapshots and get rid of the dataset support.